### PR TITLE
Initial ctl implementation

### PR DESCRIPTION
**Pull request by brooksmtownsend**
This PR implements the `ctl` subcommand which interfaces with the wasmcloud control interface.

The `ctl` subcommand itself has its own subcommands, `get`, `start` and `stop`. The full list of commands is below:
`wash ctl get hosts`
`wash ctl get inventory`
`wash ctl get claims`
`wash ctl start actor`
`wash ctl start provider`
`wash ctl stop actor`
`wash ctl stop provider`

I've tested that these commands work by commenting out the calls to the command interface in this test: https://github.com/wascc/wasmCloud/blob/main/tests/control.rs and instead manually starting / stopping actors and providers while the test waits for the action to complete. 

This PR also does the lifting of renaming `wascc` to `wasmcloud` around the `wash` project, and created a `util` module for helper functions that are shared across the cli.

For `wash ctl start {actor,provider}`, if a host is not provided then an auction will kick off for the actor/provider, then it will be scheduled on the first suitable host

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 authors = ["Brooks Townsend <brooksmtownsend@gmail.com>"]
 edition = "2018"
-repository = "https://github.com/wascc/wash"
-description = "waSCC Shell (wash) CLI tool"
+repository = "https://github.com/wasmcloud/wash"
+description = "wasmCloud Shell (wash) CLI tool"
 license = "Apache-2.0"
 readme = "README.md"
-keywords = ["webassembly", "wascc", "wash"]
+keywords = ["webassembly", "wasmCloud", "wasmcloud", "wash"]
```
Comment (autodidaddict) [inline Cargo.toml]: Pretty sure we shouldn't be using the same keyword twice. I think the rust convention is to use all lower-case and searching by keyword on crates[dot]io is case-insensitive.

File: src/ctl.rs [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+
+    /// RPC Port for connections, defaults to 4222 for local nats
+    #[structopt(short = "p", long = "rpc-port", default_value = "4222")]
+    rpc_port: String,
+
+    /// Namespace prefix for wasmCloud command interface
+    #[structopt(short = "n", long = "ns-prefix")]
+    ns_prefix: Option<String>,
+
+    /// Timeout length for RPC, defaults to 1 second
```
Comment (autodidaddict) [inline src/ctl.rs]: RPC calls should probably have a slightly higher default timeout. The "scatter gather" options from the control interface take their own independent timeouts because they potentially need to be higher than the single request-response calls.

File: src/ctl.rs [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+    Stop(StopCommand),
+}
+
+#[derive(Debug, Clone, StructOpt)]
+pub enum GetCommand {
+    /// Query lattice for running hosts
+    #[structopt(name = "hosts")]
+    Hosts(GetHostsCommand),
+
+    /// Query lattice for the inventory of a host
```
Comment (autodidaddict) [inline src/ctl.rs]: Maybe rephrase this.. it's not querying the entire lattice, it's actually probing a single host for its inventory. Maybe `Query a single host for its inventory` ?

File: src/ctl.rs [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+    #[structopt(name = "inventory")]
+    HostInventory(GetHostInventoryCommand),
+
+    /// Query lattice for claims present in the lattice
```
Comment (autodidaddict) [inline src/ctl.rs]: This is a "first responder" query. It asks the entire lattice for the claims cache and the first host that responds satisfies the query.

File: src/ctl.rs [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+
+pub async fn start_provider(cmd: StartProviderCommand) -> Result<StartProviderAck> {
+    let client = client_from_opts(cmd.opts).await?;
+    client
+        .start_provider(&cmd.host_id, &cmd.provider_ref, cmd.link_name)
+        .await
+        .map_err(convert_error)
+}
+
+pub async fn stop_provider(cmd: StopProviderCommand) -> Result<StopProviderAck> {
```
Comment (autodidaddict) [inline src/ctl.rs:343]: :smile: you were right...these control interface consumer calls are super crisp.

**Review by autodidaddict — CHANGES_REQUESTED** [review]

File: src/ctl.rs (hunk 2 above)
Comment (brooksmtownsend) [reply src/ctl.rs]: Do you think 5 seconds is a reasonable timeout length? I'm up for making this as big as we want, though if it gets much longer I may pull in some of the spinner code to show that we're working on it

**Comment by brooksmtownsend** [comment]
Marking as `do not merge` as I'm working to add in the new auction functionality to the `ctl` command before merging

File: src/ctl.rs (hunk 2 above)
Comment (autodidaddict) [reply src/ctl.rs]: I think 2 is fine, 3 max. It's a configurable option, so it's just a matter of making the default reasonable.

**Comment by autodidaddict** [comment]
> Marking as `do not merge` as I'm working to add in the new auction functionality to the `ctl` command before merging

When doing an "auction", that should probably just be included in the "start" functionality. For example, if you do `start actor bob` with no further parameters, it should try and auction for a suitable host for bob. If you supply a host name, then it can skip the auction. Similar with a provider, `start provider alfred` with no supplied host name can simply do an auction for a host name and then perform the schedule directly.

Put another way, the 2-phase process of auction-and-start is an implementation detail that we don't need to inflict on our users as additional complexity/friction.
